packedForest/src/forestTypes/unsupervisedForests/urf/fpURFBase.h

```diff
@@ -0,0 +1,151 @@
+#ifndef fpURF_h
+#define fpURF_h
+
+#include "../../../baseFunctions/fpForestBase.h"
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <unordered_map>
+#include <stdio.h>
+#include <ctime>
+#include <chrono>
+#include <cstdlib>
+#include "urfTree.h"
+#include <sys/time.h>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Sparse>
+#include <eigen3/Eigen/Core>
+using namespace Eigen;
+
+namespace fp {
+
+	template <typename T>
+		class fpURFBase : public fpForestBase<T>
+	{
+		protected:
+			std::vector<urfTree<T> > trees;
+			std::map<int, std::map<int, int> > simMat;
+			std::map<std::pair<int, int>, double> pairMat;	
+			typedef Eigen::SparseMatrix<int> spMat;
+			typedef Eigen::Triplet<int> TripType;
+			std::vector<TripType> tripletList;
+			SpMat eigenMat;
+		public:
+
+			~fpURFBase(){}
+
+			fpDisplayProgress printProgress;
+			inline void printForestType(){
+				std::cout << "This is a urf forest.\n";
+			}
+
+			inline void changeForestSize(){
+				trees.resize(fpSingleton::getSingleton().returnNumTrees());
+			}
+
+			inline void initSimMat(){
+				auto numObs = fpSingleton::getSingleton().returnNumObservations();
+				for(auto i = 0; i < numObs; ++i) {
+					std::map<int, int> init_map;
+					simMat[i] = init_map;
+				}
+			}
+			inline void growTrees(){
+#pragma omp parallel for num_threads(fpSingleton::getSingleton().returnNumThreads())
+				for(unsigned int i = 0; i < (int)trees.size(); ++i){
```

Comment:
compile time warning: `unsigned int` and `int`

Reply:
fixed in 483e171